sklearn/utils/_testing.py

```diff
@@ -710,8 +713,16 @@ def assert_run_python_script(source_code, timeout=60):
                 raise RuntimeError(
                     "script errored with output:\n%s" % e.output.decode("utf-8")
                 )
-            if out != b"":
-                raise AssertionError(out.decode("utf-8"))
+
+            out = out.decode("utf-8")
+            if re.search(pattern, out):
+                if pattern == ".+":
+                    expectation = "Expected empty output"
```

Comment:
```suggestion
                    expectation = "Expected an empty output"
```

Reply:
Actually I am not sure about this one.

Reply:
I went for "Expected no output", it seemed to have more hits in my search engine than both variants

Reply:
Fine with me